### ISSUE TITLE
False positive for NUnit1030 with `TestCaseParameters`

### ISSUE THREAD

**Issue by ehonda** [nunit/nunit.analyzers#523]
## Description

I've encountered what I believe is a false positive with the recently ([3.6.0](https://github.com/nunit/nunit.analyzers/releases/tag/3.6.0)) added [NUnit1030](https://github.com/nunit/nunit.analyzers/blob/master/documentation/NUnit1030.md). At least the code does work fine when disabling the error, and the error also does not occur when switching out `TestCaseParameters` with `TestCaseData`.

## Expected behavior

No error is produced (though I'm not sure if `TestCaseParameters` are supposed to be used this way, since they're in namespace `NUnit.Framework.Internal`, and using sub type `TestCaseData` instead works just fine).

## Reproduction

I've created a [small demo project](https://github.com/ehonda/TestCaseParametersNUnit1030/tree/main) that reproduces the false positive. It's [README](https://github.com/ehonda/TestCaseParametersNUnit1030/blob/main/README.md) has sample output.

The gist is the following:

```csharp
public class Tests
{
    private static class TestCases
    {
        public static IEnumerable<TestCaseParameters> All => new[]
        {
            new TestCaseData(1), new TestCaseData(2)
        };
    }

    [TestCaseSource(typeof(TestCases), nameof(TestCases.All))]
    public void Test(int n)
    {
        Assert.That(n, Is.GreaterThan(0));
    }
}
```

This produces error NUnit1030, although it works fine if the error is disabled, or if `IEnumerable<TestCaseParameters>` is switched out for `IEnumerable<TestCaseData>`.

**Comment by mikkelbu** [comment]
Thanks for the report @ehonda. It should also work with `TestCaseParameters`, so that is an mistake. 

https://docs.nunit.org/articles/nunit/writing-tests/attributes/testcasesource.html states
> If it is an object derived from the TestCaseDataParameters class, its properties are used to provide the test case. NUnit provides the [TestCaseData](https://docs.nunit.org/articles/nunit/writing-tests/TestCaseData.html) type for this purpose.

(here I think that `TestCaseDataParameters` is a typo - it should have said `TestCaseParameters`)

I'll see if I can get a fix out one of the following days (and update the documentation).

**Comment by manfred-brands** [comment]
@mikkelbu Since I wrote it, I will create the fix for the source code.
